js/aax.js

```diff
@@ -3238,6 +3238,46 @@ module.exports = class aax extends Exchange {
         };
     }
 
+    async fetchTransactionFees (codes = undefined, params = {}) {
+        /**
+         * @method
+         * @name aax#fetchTransactionFees
+         * @description fetch transaction fees
+         * @see https://www.aax.com/apidoc/index.html#get-currencies
+         * @param {[string]|undefined} codes list of unified currency codes
+         * @param {object} params extra parameters specific to the aax api endpoint
+         * @returns {object} a list of [fee structures]{@link https://docs.ccxt.com/en/latest/manual.html#fee-structure}
+         */
+        await this.loadMarkets ();
+        const currencyKeys = Object.keys (this.currencies);
```

Comment:
You should call the exchange API instead of emulating the method using this.currencies, so that you get the most up to date information

Reply:
Hmmm, this is in contrast with what @carlosmiei told me. https://github.com/ccxt/ccxt/pull/15439#discussion_r1008664868

Reply:
I'm using this same [logic used here](https://github.com/ccxt/ccxt/pull/11942#issuecomment-1038194703)

Reply:
@kroitor Can you give your input please, seeing as [your explanation](https://github.com/ccxt/ccxt/pull/11942#issuecomment-1038194703) is in contrast to [Carlos explanation](https://github.com/ccxt/ccxt/pull/15439#discussion_r1008664868). Also @carlosmiei

Reply:
@samgermain I agree with the param filtering but not so much with "so that you get the most up to date information", these fees are not that volatile IMO

Reply:
@AlgoQ @samgermain ok Igor's comment seems to indicate we should not use the cached `currencies` here, my bad, @AlgoQ can you update it?

Reply:
@AlgoQ @samgermain @carlosmiei @kroitor 
I think we can do the following and catch two rabbits:
- if `fetchCurrencies` use the same endpoint, then in `fDWF` shouldn't separately call that endpoint, instead, in `fDWF` we should call `await this.fetchCurrencies()` and then use `this.currencies` in `fDWF`. The advantage will be that, as long as it calls the same endpoint, we get benefit of re-loading the currencies list too in addition (so, in the meantime if there has been added new coins, they will also be propagated in exchange class)

Reply:
> @AlgoQ @samgermain @carlosmiei @kroitor I think we can do the following:
> 
>     * if `fetchCurrencies` use the same endpoint, then in `fDWF` shouldn't separately call that endpoint, instead, in `fDWF` we should call `await this.fetchCurrencies()` and then use `this.currencies` in `fDWF`. The advantage will be that, as long as it calls the same endpoint, we get benefit of re-loading the currencies list too in addition (so, in the meantime if there has been added new coins, they will also be propagated in exchange class).
>       so, we catch two rabbits with one shot.

@kroitor Do you agree?